# Japan Population Analysis

In this project, I analyze population data of Japan across different prefectures.

The goal is to understand how population is distributed and apply statistical concepts such as mean, median, and measures of spread on real-world data.

This project builds on basic statistical concepts and applies them to a real dataset.

In [2]:
import pandas as pd
import matplotlib.pyplot as plt

## Loading the Dataset

The dataset is loaded using pandas. It contains population data categorized by prefecture, age group, gender, and year.

In [3]:
df = pd.read_csv("data/japanese_census.csv")
df.head()

,Unnamed: 0,Year,Prefecture,Age Lower Bound,Age Upper Bound,Gender,Population
0,0,1920,Hokkaido,0,4.0,Male,190044
1,1,1920,Hokkaido,0,4.0,Female,186632
2,2,1920,Hokkaido,5,9.0,Male,161814
3,3,1920,Hokkaido,5,9.0,Female,157556
4,4,1920,Hokkaido,10,14.0,Male,138018


## Understanding the Dataset

Before analysis, it is important to understand the structure of the data, including available columns.

In [4]:
df.columns

Index(['Unnamed: 0', 'Year', 'Prefecture', 'Age Lower Bound',
       'Age Upper Bound', 'Gender', 'Population'],
      dtype='str')

##  Data Cleaning

The dataset contains unnecessary columns and multiple breakdowns such as age, gender, year. 

To make the data usable:
- Remove unnecessary columns  
- Filter data for the latest year  
- Aggregate population for each prefecture  

In [5]:
df = df.drop(columns=['Unnamed: 0'], errors='ignore') #since the column would be deleted and if the code ran again it will give error
df = df[df['Year'] == df['Year'].max()]
df['Prefecture'] = df['Prefecture'].replace('Tokyo Metropolis', 'Tokyo Prefecture') # as tokyo prefecture is commonly named tokyo metropolis or tokyo-to
df_grouped = df.groupby('Prefecture')['Population'].sum().reset_index()
df_grouped.head()

,Prefecture,Population
0,Aichi Prefecture,7401952
1,Akita Prefecture,1014579
2,Aomori Prefecture,1297015
3,Chiba Prefecture,6126343
4,Ehime Prefecture,1362407


### Note on Dataset

The dataset contains 46 administrative regions instead of the expected 47 prefectures of Japan.

This may be due to how the dataset categorizes regions. The analysis will proceed using the available data.

### Exploratory Data Analysis

Before applying statistical measures, it is important to explore the cleaned dataset and understand its structure, size, and quality.

In [ ]:
df_grouped.shape 

(46, 2)

Data has 46 prefectures and 2 columns.

In [ ]:
df_grouped.dtypes 

Prefecture      str
Population    int64
dtype: object

The dataset contains:
- Prefecture as a string/object column
- Population as an integer column

This confirms that the population data is numeric and suitable for statistical analysis.

In [ ]:
df_grouped.isnull().sum() 

Prefecture    0
Population    0
dtype: int64

No missing values were found in the cleaned dataset.

This means the dataset is complete and ready for analysis.

In [11]:
df_grouped.describe()

,Population
count,4.600000e+01
mean,2.731326e+06
std,2.720719e+06
min,5.690780e+05
25%,1.124096e+06
50%,1.703647e+06
75%,2.748845e+06
max,1.325780e+07


The summary statistics provide an overview of the population distribution across prefectures.

Key observations:
- The average population is around 2.7 million
- The median population is lower than the mean
- This suggests that a few highly populated prefectures increase the average significantly
- The large standard deviation indicates high variability in population distribution

## Central Tendency Analysis

Central tendency measures help identify the typical or central value of a dataset.

In this analysis, the following measures will be explored:
- Mean
- Median

In [18]:
mean_population = df_grouped['Population'].mean()
round(mean_population, 2)

np.float64(2731325.8)

### Mean Population

The mean population across prefectures is approximately 2.73 million.

Since the mean is affected by extremely large values, highly populated prefectures such as Tokyo increase the overall average population.

In [17]:
median_population = df_grouped['Population'].median()
round(median_population)

1703647

### Median Population

The median population is approximately 1.7 million.

The median is lower than the mean population, suggesting that a few highly populated prefectures increase the average significantly.

This indicates that the population distribution is right-skewed.

## Comparison of Mean and Median

The mean population is significantly higher than the median population.

This indicates that the dataset contains a few prefectures with extremely high populations, which increase the average population value.

Therefore, the median provides a more stable representation of a typical prefecture population.